macosx/kext/AudioEngine.cpp

```diff
@@ -460,13 +461,67 @@ IOAudioStream *kXAudioEngine::createNewAudioStream(int chn,IOAudioStreamDirectio
 			if(sampleBuffer && sampleBufferSize)
 				audioStream->setSampleBuffer(sampleBuffer, sampleBufferSize);
             
-				// This device only allows a single format and a choice of 2 different sample rates
-            rate.fraction = 0;
-            rate.whole = sampling_rate;
-            audioStream->addAvailableFormat(&format, &rate, &rate);
-            
-				// Finally, the IOAudioStream's current format needs to be indicated
-            audioStream->setFormat(&format);
+            if(hw->is_10k2){
+                rate.fraction = 0;
+                rate.whole = sampling_rate;
+                format.fBitDepth = 24;
+                format.fBitWidth = 32;
+                audioStream->addAvailableFormat(&format, &rate, &rate);
+                
+                // Finally, the IOAudioStream's current format needs to be indicated
+                audioStream->setFormat(&format);
+                
+                rate.whole = 44100;
+                format.fBitDepth = 24;
+                format.fBitWidth = 32;
```

Comment:
True 24/32-bit support requires b2b plugins in the DSP + extra handling of audio data.